hazelcast/src/hazelcast/client/xml_impl.cpp

```
@@ -0,0 +1,1317 @@
+  const boost::property_tree::ptree& node)
+{
+    handle_near_cache_node(node);
+}
+void
+client_dom_config_processor::handle_near_cache_node(
+  const boost::property_tree::ptree& node) const
+{
+    std::string name;
+    try {
+        name = get_attribute(node, "name");
+    } catch (const boost::exception& e) {//if name is not explicitly specified then name = default
+        name = "default";
+    }
+    hazelcast::client::config::near_cache_config near_cache_config(name);
+    for (auto& pair : node) {
+        auto child = pair_to_node(pair.first, pair.second);
+        std::string node_name = pair.first;
+        if (matches("time-to-live-seconds", node_name)) {
+            near_cache_config.set_time_to_live_seconds(
+              get_integer_value(pair.first, pair.second.data()));
+        } else if (matches("max-idle-seconds", node_name)) {
+            near_cache_config.set_max_idle_seconds(
+              get_integer_value(pair.first, pair.second.data()));
+        } else if (matches("in-memory-format", node_name)) { // NO NATIVE ??
+            if (pair.second.data() == "BINARY") {
+                near_cache_config.set_in_memory_format(
+                  hazelcast::client::config::in_memory_format::BINARY);
+            } else if (pair.second.data() == "OBJECT") {
+                near_cache_config.set_in_memory_format(
+                  hazelcast::client::config::in_memory_format::OBJECT);
+            }
+        } else if (matches("invalidate-on-change", node_name)) {
+            near_cache_config.set_invalidate_on_change(
+              get_bool_value(pair.second.data()));
+        } else if (matches("local-update-policy", node_name)) {
+            if (pair.second.data() == "CACHE") {
+                near_cache_config.set_local_update_policy(
+                  hazelcast::client::config::near_cache_config::CACHE);
+            } else if (pair.second.data() == "INVALIDATE") {
+                near_cache_config.set_local_update_policy(
```

If 'local-update-policy' is wrong it should throw an exception according to java implementation.